### PR TITLE
Broadcasting value updates

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
Addresses #680 and #687.

name unicity is "guaranteed" by PropertyType.broadcast and Gui.broadcast()

### CODE REVIEW

File: gui/packaging/taipy-gui.d.ts [hunk 1]
```diff
@@ -138,6 +138,17 @@ export declare const createSendActionNameAction: (name: string | undefined, valu
  * @returns The action fed to the reducer.
  */
 export declare const createRequestDataUpdateAction: (name: string | undefined, id: string | undefined, columns: string[], pageKey: string, payload: Record<string, unknown>, allData?: boolean, library?: string) => Action;
+/**
+ * Creates a *request update* `Action` that will be used to update the `Context`.
```
Comment (FabienLelaquais) [inline gui/packaging/taipy-gui.d.ts]: That would be "Create".
Do you want cross-refs on the classes as well? (I bet you do)

File: gui/packaging/taipy-gui.d.ts [hunk 2]
```diff
@@ -138,6 +138,17 @@ export declare const createSendActionNameAction: (name: string | undefined, valu
...
+ *
+ * This action will generate an update of the elements holding the variables named
+ * *names* on the front-end.
+ * @param id - The identifier of the visual element.
```
Comment (FabienLelaquais) [inline gui/packaging/taipy-gui.d.ts:146]: What if `id` is undefined?

File: gui/packaging/taipy-gui.d.ts [hunk 3]
```diff
@@ -138,6 +138,17 @@ export declare const createSendActionNameAction: (name: string | undefined, valu
...
+ * @param names - The names of the requested variable as received in updateVarName and/or updateVars properties.
```
Comment (FabienLelaquais) [inline gui/packaging/taipy-gui.d.ts]: "requested variable" -> "requested variables"
"updateVarName and/or updateVars" -> "\`updateVarName()\` and \`updateVars()\`"
"properties"??

File: gui/packaging/taipy-gui.d.ts [hunk 4]
```diff
@@ -138,6 +138,17 @@ export declare const createSendActionNameAction: (name: string | undefined, valu
...
+ * @param forceRefresh - Should Taipy recalculate the variables or use the current values.
```
Comment (FabienLelaquais) [inline gui/packaging/taipy-gui.d.ts]: Is "recalculate the variables" equivalent to "re-evaluate"? And better?

File: gui/src/context/taipyReducers.ts [hunk 5]
```diff
@@ -649,12 +656,23 @@ export const createRequestDataUpdateAction = (
     };
 };
 
-export const createRequestUpdateAction = (id: string | undefined, names: string[]): TaipyAction => ({
+/**
+ * Creates a *request update* `Action` that will be used to update the `Context`.
```
Comment (FabienLelaquais) [inline gui/src/context/taipyReducers.ts]: That would be "Create".
Do you want cross-refs on the classes as well? (I bet you do)

File: gui/src/context/taipyReducers.ts [hunk 6]
```diff
@@ -649,12 +656,23 @@ export const createRequestDataUpdateAction = (
...
+ *
+ * This action will generate an update of the elements holding the variables named
+ * *names* on the front-end.
+ * @param id - The identifier of the visual element.
```
Comment (FabienLelaquais) [inline gui/src/context/taipyReducers.ts:664]: What if *id* is undefined?

File: gui/src/context/taipyReducers.ts [hunk 7]
```diff
@@ -649,12 +656,23 @@ export const createRequestDataUpdateAction = (
...
+ * @param names - The names of the requested variable as received in updateVarName and/or updateVars properties.
```
Comment (FabienLelaquais) [inline gui/src/context/taipyReducers.ts]: "requested variable" -> "requested variables"
"updateVarName and/or updateVars" -> "`updateVarName()` and `updateVars()`"
"properties"??

File: gui/src/context/taipyReducers.ts [hunk 8]
```diff
@@ -649,12 +656,23 @@ export const createRequestDataUpdateAction = (
...
+ * @param forceRefresh - Should Taipy recalculate the variables or use the current values
```
Comment (FabienLelaquais) [inline gui/src/context/taipyReducers.ts]: Is "recalculate the variables" equivalent to "re-evaluate"? And better?

File: src/taipy/gui/extension/library.py [hunk 9]
```diff
@@ -48,8 +48,15 @@ def __init__(
                 "my_property_name", then this property is referred to as "myPropertyName" in the
                 JavaScript code.
         """
-        self.property_type = property_type
         self.default_value = default_value
+        if property_type == PropertyType.broadcast:
+            if isinstance(default_value, str):
+                self.default_value = _get_broadcast_var_name(default_value)
+            else:
+                warnings.warn("Element property with type 'broadcast' should define a string default value")
```
Comment (FabienLelaquais) [inline src/taipy/gui/extension/library.py]: should -> must

**Review by dinhlongviolin1 — APPROVED** [review]
Amazingly done! Could not find a thing that I don't like

File: src/taipy/gui/gui.py [hunk 10]
```diff
@@ -1444,9 +1470,19 @@ def __bind_local_func(self, name: str):
     def load_config(self, config: Config) -> None:
         self._config._load(config)
 
+    def broadcast(self, name: str, value: t.Any):
+        """
+        Send a new value for a variable to all connected clients.
+
+        Arguments:
+            name: The name of the variable to update/create.
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui.py]: update or create

File: src/taipy/gui/gui.py [hunk 11]
```diff
@@ -1444,9 +1470,19 @@ def __bind_local_func(self, name: str):
...
+            value: The value (should be serializable to json format)
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui.py]: should -> must

**Review by FabienLelaquais — APPROVED** [review]
Minor doc-related comments.
Excellent!

File: gui/packaging/taipy-gui.d.ts (hunk 1 above)
Comment (FredLL-Avaiga) [reply gui/packaging/taipy-gui.d.ts]: what classes ?

File: gui/packaging/taipy-gui.d.ts (hunk 2 above)
Comment (FredLL-Avaiga) [reply gui/packaging/taipy-gui.d.ts:146]: then it is undefined :-)

File: gui/packaging/taipy-gui.d.ts (hunk 3 above)
Comment (FredLL-Avaiga) [reply gui/packaging/taipy-gui.d.ts]: properties are named updateVarName & updateVars

**Pull request merged**